cocos/base/ObjectFactory.cpp

```diff
@@ -24,65 +24,27 @@ THE SOFTWARE.
 ****************************************************************************/
 
 #include "base/ObjectFactory.h"
-#include <functional>
 #include <utility>
 
 
 NS_CC_BEGIN
 
-ObjectFactory::TInfo::TInfo()
-:_class("")
-,_fun(nullptr)
-,_func(nullptr)
-{
-}
-
 ObjectFactory::TInfo::TInfo(const std::string& type, Instance ins)
 :_class(type)
 ,_fun(ins)
-,_func(nullptr)
 {
     ObjectFactory::getInstance()->registerType(*this);
 }
 
 ObjectFactory::TInfo::TInfo(const std::string& type, const InstanceFunc& ins)
-    :_class(type)
-    ,_fun(nullptr)
-    ,_func(ins)
+:_class(type)
+,_func(ins)
 {
     ObjectFactory::getInstance()->registerType(*this);
 }
 
-ObjectFactory::TInfo::TInfo(const TInfo &t)
-{
-    _class = t._class;
-    _fun = t._fun;
-    _func = t._func;
-}
-
-ObjectFactory::TInfo::~TInfo()
-{
-   _class = "";
-   _fun = nullptr;
-   _func = nullptr;
-}
-
-ObjectFactory::TInfo& ObjectFactory::TInfo::operator= (const TInfo &t)
-{
-    _class = t._class;
-    _fun = t._fun;
-    _func = t._func;
-    return *this;
-}
-
-
 ObjectFactory* ObjectFactory::_sharedFactory = nullptr;
 
-ObjectFactory::ObjectFactory()
-{
-
-}
-
 ObjectFactory::~ObjectFactory()
```

Comment:
Not needed.
Default destructor do the same

Reply:
It's declared as virtual in header file. But we can simplify it with `virtual ~ObjectFactory() = default;`.

Reply:
But it doesn't provide any other virtual method, so I am not sure why `virtual` is needed in destructor.

Reply:
We may declare this class as final and remove virtual destructor or if that fails leave virtual destructor and add =default